### ISSUE TITLE
Implement User ID tracking for Google Analytics

### ISSUE THREAD

**Issue by CourtneyRay** [openstax/bit#268]
### Description
Implement Google Analytics's User ID tracking feature. Accounts needs to communicate to GA a unique ID per each user's account. When logged in, the users' activity will be tracked across devices and sessions over time. It will give a clearer picture of users' behavior, what is working, what they like, and what drives them to take certain actions.

Courtney has already enabled User ID tracking in the OSWeb properties. See image below for implementation requirements in tracking code.

![Screen Shot 2019-02-18 at 12.54.42 PM.png](https://images.zenhubusercontent.com/5b7ed2d5bc512f725222c253/f4eb25f3-8d83-4bb9-a0e2-3ed893e14ae6)

### Acceptance
- Accounts communicates an assigned user number to Google Analytics (tracking properties UA-73668038-1 and UA-73668038-2)
- user tracking shows up in Google Analytics (tested after implementation - or can deploy to dev/QA first and check dev/QA analytics properties)
- A blurb is added to our privacy policy describing Google Analytics as one of the platforms we use that collects data, and that gives contact information on how a user can a) opt out of data collection going forward, and/or b) have their historical tracked data erased. Can view https://www.unitedway.org/privacy-policy as an example (though we need to tailor a bit according to our own practices)
- After this is implemented, a footer appears the first time a user lands on the site while logged in or the first time they log in while on the site. The footer explains that we collect data and allows the user to click a button to accept, or click another button to "learn more." The "learn more" lands on the privacy policy. See www.unitedway.org as an example.

### Impact
We need to implement this change in order to measure key success metrics, including what combinations of engagement over time leads users take leading up to adoption, accessing resources, using REX, returning to the site, and other key actions.

### Documentation and info
Set up User ID: https://support.google.com/analytics/answer/3123666
About User ID: https://support.google.com/analytics/answer/3123662?hl=en
Courtney's summary of User ID, benefits, and risks: https://docs.google.com/document/d/1tdPHkBCfE_tIMoAxqUgXZiYaZZFf-Pr-P4u8cF2V0OM/edit#

* **Severity**: [X] High [ ] Medium  [ ] Low

**Comment by allisonhowell** [comment]
@victoria-cooke We still need to work out copy, but per your comment, we can go ahead and start design for this card.

**Comment by victoria-cooke** [comment]
@allisonhowell User ID tracking options for review: https://invis.io/JVSHBM1XTB6

**Comment by victoria-cooke** [comment]
@allisonhowell Updated design: https://invis.io/JVSHBM1XTB6#/369678511_20190612_BIT_IDTracking_VCooke_v3-01

**Comment by allisonhowell** [comment]
@mmmcglone Roy, Victoria & I discussed this in scrum and came up with these decisions:

- The cookie is only used when someone logs in, so we will only display the message to logged in users - either if they arrive on the page logged in or when they log in. @victoria-cooke @RoyEJohnson Will we want some kind of popup to appear when they log in rather than the footer?

- The message will be displayed on the landing page in the session. If a user navigates away from the landing page it will disappear. 

- It will continue to display each time a logged in user goes to openstax.org, until they click that they've "got it."

I believe the outstanding items are 1) Should we add an email address to the privacy policy, and 2) What should the final copy be?

@victoria-cooke @RoyEJohnson Did I miss anything from our conversation or does that cover it?

**Comment by mmmcglone** [comment]
@allisonhowell thanks for filling me in! I've followed up with Beth and apparently she has already asked Veta (part of the general counsel team with Rice) to look into state-specific standards re: contact information in privacy policy. Word is still out on whether we'd need to communicate a change like that to users. But I think it's save to say we'll need to added a contact email address to the privacy policy.

**Comment by victoria-cooke** [comment]
@allisonhowell Yes to the pop-up instead of the footer, and it looks like you've covered everything.

**Comment by allisonhowell** [comment]
Allison will update into two separate cards

**Comment by Brooke-Fail** [comment]
Verified this is not in QA or Staging at the moment.

**Comment by Brooke-Fail** [comment]
Verified that this is not in Production at the moment.

**Comment by Brooke-Fail** [comment]
Will need to be verified after released.
